tap_mysql/client.py

```diff
@@ -170,6 +214,167 @@ def get_schema_names(self, engine: Engine, inspected: Inspector) -> list[str]:
+            for column in columns:
+                column_name = column["Field"]
+                is_nullable = column["Null"] == "YES"
+                jsonschema_type: dict = self.to_jsonschema_type(column["Type"])
+                table_schema.append(
+                    th.Property(
+                        name=column_name,
+                        wrapped=th.CustomType(jsonschema_type),
+                        required=not is_nullable,
+                    ),
+                )
+        schema = table_schema.to_dict()
+
+        # Initialize available replication methods
+        addl_replication_methods: list[str] = [""]  # By default an empty list.
+        # Notes regarding replication methods:
+        # - 'INCREMENTAL' replication must be enabled by the user by specifying
+        #   a replication_key value.
+        # - 'LOG_BASED' replication must be enabled by the developer, according
+        #   to source-specific implementation capabilities.
+        replication_method = next(reversed(["FULL_TABLE", *addl_replication_methods]))
+
+        # Create the catalog entry object
+        return CatalogEntry(
+            tap_stream_id=unique_stream_id,
+            stream=unique_stream_id,
+            table=table_name,
+            key_properties=None,
+            schema=Schema.from_dict(schema),
+            is_view=is_view,
+            replication_method=replication_method,
+            metadata=MetadataMapping.get_standard_metadata(
+                schema_name=schema_name,
+                schema=schema,
+                replication_method=replication_method,
+                key_properties=None,
+                valid_replication_keys=None,  # Must be defined by user
+            ),
+            database=None,  # Expects single-database context
+            row_count=None,
+            stream_alias=None,
+            replication_key=None,  # Must be defined by user
+        )
+
+    def get_sqlalchemy_type(self, col_meta_type: str) -> sqlalchemy.Column:
+        """Return a SQLAlchemy type object for the given SQL type.
+
+        Used ischema_names so we don't have to manually map all types.
+        """
+        dialect = sqlalchemy.dialects.mysql.base.dialect()  # type: ignore[attr-defined]
+        ischema_names = dialect.ischema_names
+        # Example varchar(97)
+        type_info = col_meta_type.split("(")
+        base_type_name = type_info[0].split(" ")[0]  # bigint unsigned
+        type_args = (
+            type_info[1].split(" ")[0].rstrip(")") if len(type_info) > 1 else None
+        )  # decimal(25,4) unsigned should work
+
+        if base_type_name in {"enum", "set"}:
+            self.logger.warning(
+                "Enum and Set types not supported for col_meta_type=%s. "
+                "Using varchar instead.",
+                col_meta_type,
+            )
+            base_type_name = "varchar"
+            type_args = None
+
+        type_class = ischema_names.get(base_type_name.lower())
```

Comment:
ischema_names

Reply:
Can you expand on what's the issue?

Reply:
Parsing SQL DDL/DML seems like the job we have `sqlalchemy` doing. In this `get_sqlalchemy_type` function I'm doing a bunch of string splitting / stripping which is fine, but I'd rather punt this work to `sqlalchemy` 

Curious if anyone has a better idea here!

Reply:
So what would that look in pseudo-sqlalchemy? i.e. imagining the function signatures and names

Reply:
Ideally it'd be like this https://github.com/meltano/sdk/blob/ae7614e152f9f6a479615b6fe42f48baf4e97494/singer_sdk/connectors/sql.py#L613

so something like `inspector.get_columns(view_name, schema_name)` , and we wouldn't have to talk about anything regarding sql parsing (that's what the mysql dialect does for us in most situations here https://github.com/sqlalchemy/sqlalchemy/blob/5c88498ba8ba5f7c524d5aca130e5a59a8940766/lib/sqlalchemy/dialects/mysql/reflection.py#L329 )

In "pseudo-sqlalchemy" for this specific function I was thinking something like

```sqlalchemy.reflection.column_ddl_to_column("varchar(48)") -> sqlalchemy.Column() ```

Reply:
SQLGlot could probably help us here, but at this point we're probably at the "good enough" stage 🤷

Reply:
Yeah, I don't know of any SQLAlchemy APIs to parse a column type string into a TypeEngine instance 👍